modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/MediaHandler.java

```diff
@@ -153,59 +170,38 @@ public WebResponse handle( final WebRequest webRequest, final WebResponse webRes
         final String fingerprint = matcher.group( "fingerprint" );
         final String restPath = matcher.group( "restPath" );
 
-        if ( !defaultContextPathVerifier.verify( webRequest ) )
-        {
-            throw WebException.notFound( "Not a valid media url pattern" );
-        }
-
-        verifyMediaScope( matcher.group( "context" ), repositoryId, branch, webRequest );
+        verifyMediaScope( matcher.group( "context" ), repositoryId, branch, portalRequest );
+        verifyAccessByBranch( branch );
 
-        final PortalRequest portalRequest = createPortalRequest( webRequest, repositoryId, branch );
+        portalRequest.setRepositoryId( repositoryId );
```

Comment:
This is a bit shaky. Overriding request repo/branch fools downstream logic what was actually requested. 
Maybe better to read all needed parameters from request and provide them as arguments to downstream methods instead of draging the entire request

Reply:
It was done for a `/api/...` request